src/ConnectedMode.UnitTests/Binding/BindingInfoProviderTests.cs

```diff
@@ -0,0 +1,124 @@
+        {
+            var fileSytem = new MockFileSystem();
+            fileSytem.AddDirectory("C:\\Bindings");
+            fileSytem.AddDirectory("C:\\Bindings\\Binding1");
+            fileSytem.AddDirectory("C:\\Bindings\\Binding2");
+
+            var file1 = CreateFileData("https://sonarqube.somedomain.com", null, "projectKey1");
+            var file2 = CreateFileData("https://sonarcloud.io", "organisation2", "projectKey2");
+
+            fileSytem.AddFile("C:\\Bindings\\Binding1\\binding.config", file1);
+            fileSytem.AddFile("C:\\Bindings\\Binding2\\binding.config", file2);
+
+            var testSubject = CreateTestSubject(fileSytem);
+
+            var result = testSubject.GetExistingBindings();
+
+            result.Should().HaveCount(2);
+            result[0].ServerUri.Should().Be("https://sonarqube.somedomain.com");
+            result[0].Organization.Should().BeNull();
+            result[0].ProjectKey.Should().Be("projectKey1");
+            result[1].ServerUri.Should().Be("https://sonarcloud.io");
+            result[1].Organization.Should().BeNull("organisation2");
+            result[1].ProjectKey.Should().Be("projectKey2");
+        }
+
+        [TestMethod]
+        public void GetExistingBindings_BindingConfigMissing_SkipFile()
+        {
+            var fileSytem = new MockFileSystem();
+            fileSytem.AddDirectory("C:\\Bindings");
+            fileSytem.AddDirectory("C:\\Bindings\\Binding1");
+            fileSytem.AddDirectory("C:\\Bindings\\Binding2");
+
+            var file = CreateFileData("https://sonarcloud.io", "organisation", "projectKey");
+
+            fileSytem.AddFile("C:\\Bindings\\Binding2\\binding.config", file);
+
+            var testSubject = CreateTestSubject(fileSytem);
+
+            var result = testSubject.GetExistingBindings();
+
+            result.Should().HaveCount(1);
+        }
+
+        private static IUnintrusiveBindingPathProvider CreateUnintrusiveBindingPathProvider()
+        {
+            var unintrusiveBindingPathProvider = new Mock<IUnintrusiveBindingPathProvider>();
+            unintrusiveBindingPathProvider.SetupGet(u => u.SLVSRootBindingFolder).Returns("C:\\Bindings");
+            return unintrusiveBindingPathProvider.Object;
+        }
+
+        private static BindingInfoProvider CreateTestSubject(MockFileSystem fileSytem)
+        {
+            var unintrusiveBindingPathProvider = CreateUnintrusiveBindingPathProvider();
+
+            var testSubject = new BindingInfoProvider(unintrusiveBindingPathProvider, fileSytem);
+            return testSubject;
+        }
+
+        private static MockFileData CreateFileData(string serverUri, string organization, string projectKey)
+        {
+            var content = organization != null
+                ? $"{{\"ServerUri\":\"{serverUri}\",\"Organization\":{{\"Key\":\"{organization}\",\"Name\":\"\"}},\"ProjectKey\":\"{projectKey}\",\"ProjectName\":\"\"}}"
```

Comment:
For the sake of being as close to reality as possible, you could serialize full `BoundSonarQubeProject` model here